lib/pubsub/topic.js

```diff
@@ -63,6 +63,27 @@ function Topic(pubsub, options) {
 }
 
 /**
+ * Format a message object as the upstream API expects it.
+ *
+ * @private
+ *
+ * @return {object}
+ */
+Topic.formatMessage_ = function(message) {
+  if (!message.data) {
+    message = {
+      data: new Buffer(message).toString('base64')
+    };
+  }
+
+  if (!util.is(message.data, 'string') && !util.is(message.data, 'buffer')) {
+    message.data = JSON.stringify(message.data);
```

Comment:
Going to think out loud here in case anyone can help me out with this.

This is to format a message before being published to the API. The API expects a base64 encoded string value for message contents (`message.data`). What I'm struggling with is:

- how to know if the user gave us a value already encoded,
- base64 encoding a number (`new Buffer(JSON.stringify(3))`?)
- base64 encoding an object (`new Buffer(JSON.stringify({ userId: 3 })`?)

Please let me know if I've said something silly.